src/herder/HerderImpl.cpp

```diff
@@ -825,6 +825,36 @@ HerderImpl::getMaxSeqInPendingTxs(AccountID const& acc)
     return mTransactionQueue.getAccountTransactionQueueInfo(acc).mMaxSeq;
 }
 
+void
+HerderImpl::setInSyncAndTriggerNextLedger()
+{
+    // We either have not set trigger timer, or we're in the
+    // middle of a consensus round. Either way, we do not want
+    // to trigger ledger, as the node is already making progress
+    if (mTriggerTimer.seq() > 0)
+    {
+        CLOG(DEBUG, "Herder") << "Skipping setInSyncAndTriggerNextLedger: "
+                                 "trigger timer already set";
+        return;
+    }
+
+    // Bring Herder and LM in sync in case they aren't
```

Comment:
Yikes, it ends up being not as straightforward as I thought 😅 

As far as I can tell, `restoreSCPState` doesn't do what you want when lcl is genesis (so starting fresh) as it no-ops in that case.

More importantly, this code is still unsafe, imagine the following sequence takes place:
1. node waits for consensus
2. node receives enough messages to externalize
3. (node may transition to catchup)
4. node goes out of sync
5. we invoke the `manualclose` endpoint

At this point
* `restoreSCPState` will perform potentially unsafe operations (overrides a bunch of SCP related fields, potentially rolls back certain things in SCP itself)
* `mLedgerManager.bootstrap` potentially sets the wrong state (if catching up)

Now, do we need to actually do this?
* for `mLedgerManager`
   * if it's "catching up" we have to no-op
   * otherwise, we can call `mLedgerManager.bootstrap()` (that will either no-op or perform the BOOTING->SYNCED transition) - maybe that function should be renamed `forceTransitionToSynced`?
* for SCP, the only thing we care about is to make sure that we're nominating against the local state (so `lcl`) so that it can properly validate values. I don't think we need to mess with "tracking"
   * basically in this particular situation, we can bypass the check `!mHerderSCPDriver.trackingSCP()` ; we still need to ensure that `mLedgerManager.isSynced()` is true though
   * we could call a variation of `restoreSCP` on startup, but that is a minor improvement. This would make it safer if we're using this endpoint when restarting the network from a state that's not genesis
       * we'd remove the condition on `FORCE_SCP` (ie, always restore when not on genesis)
       * make it no-op if the node is not validating
       * always restores the SCP state
       * skips setting tracking state when `FORCE_SCP` is not set (doesn't call `HerderSCPDriver::restoreSCPState`)

Reply:
Updated.

As discussed offline, I did not change `restoreSCPState`, as it already handles non-genesis ledgers properly.